### PR TITLE
Removed Freeze Edge Case

### PR DESCRIPTION
**Pull request by etwest**
Switched to wait_for to fix the case where the tests were freezing.

### CODE REVIEW

File: include/buffer_control_block.h [hunk 1]
```diff
@@ -22,8 +22,6 @@ typedef uint64_t File_Pointer;
 class BufferControlBlock {
 private:
   buffer_id_t id;
-  // busy lock
-  std::mutex mtx;
```
Comment (Victor-C-Zhang) [inline include/buffer_control_block.h:26]: I was afraid this was the issue. Is it possible to instead just enforce a global order on acquiring these mutexes? For instance, the canonical BFS order starting from root.

File: include/circular_queue.h [hunk 2]
```diff
@@ -6,10 +6,10 @@
 #include <utility>
 
 struct queue_elm {
-	bool dirty;    // is this queue element yet to be processed by sketching (if so do not overwrite)
-	bool touched;  // have we peeked at this item (if so do not peek it again)
-	uint32_t size; // the size of this data element (in bytes)
-	char *data;    // a pointer to the data
+	volatile bool dirty;    // is this queue element yet to be processed by sketching (if so do not overwrite)
```
Comment (Victor-C-Zhang) [inline include/circular_queue.h:9]: why not just have locking on the queue?

File: src/circular_queue.cpp [hunk 3]
```diff
@@ -39,7 +39,7 @@ void CircularQueue::push(char *elm, int size) {
 	while(true) {
 		std::unique_lock<std::mutex> lk(write_lock);
 		// printf("CQ: push: wait on not-full. full() = %s\n", (full())? "true" : "false");
-		cirq_full.wait(lk, [this]{return !full();});
+		cirq_full.wait_for(lk, std::chrono::seconds(2), [this]{return !full();});
```
Comment (Victor-C-Zhang) [inline src/circular_queue.cpp:42]: hold on, why does pure wait cause an error?

**Review by Victor-C-Zhang — APPROVED** [review]
I'm deeply uncomfortable about these changes, but in the interest of time let's get em in.

File: include/buffer_control_block.h (hunk 1 above)
Comment (etwest) [reply include/buffer_control_block.h:26]: For posterity: not the issue just removing dead code.

File: include/circular_queue.h (hunk 2 above)
Comment (etwest) [reply include/circular_queue.h:9]: We do. This was a bit of a throw things at the wall change. However, we have two separate mutexes protecting the queue so maybe there's something weird there.

File: src/circular_queue.cpp (hunk 3 above)
Comment (etwest) [reply src/circular_queue.cpp:42]: In some small number of cases "something" was happening which made both the push and peek wait on the queue because of the condition variables. Then the entire process ground to a halt. The wait_for enforces that the longest this could possibly happen for is 2 seconds.

Comment (etwest) [reply src/circular_queue.cpp:42]: Any additional cost because of wait_for should be negligible due to the longer timeout.
